pkg/asset/installconfig/baremetal/baremetal.go

```diff
@@ -0,0 +1,165 @@
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &libvirtURI)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "External bridge",
+				Help:    "External bridge is used for external communication.",
+				Default: baremetaldefaults.ExternalBridge,
+			},
+			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
+		},
+	}, &externalBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning bridge",
+				Help:    "Provisioning bridge is used to provision machines.",
+				Default: baremetaldefaults.ProvisioningBridge,
+			},
+			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
+		},
+	}, &provisioningBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "API VIP",
+				Help:    "The VIP to be used for internal API communication. If blank, the address will be looked up from DNS.",
+				Default: baremetaldefaults.APIVIP,
+			},
+			Validate: apiVIPValidator,
+		},
+	}, &apiVIP)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Ingress VIP",
+				Help:    "The VIP to be used for ingress. If blank, the address will be looked up from DNS.",
+				Default: baremetaldefaults.IngressVIP,
+			},
+			Validate: ingressVIPValidator,
+		},
+	}, &ingressVIP)
+	if err != nil {
+		return nil, err
+	}
+
+	// Keep prompting for hosts
+	for {
+		var host *baremetal.Host
+		host, err = Host()
+		host.Role = "master" // FIXME(stbenjam): Support workers
+
+		if err != nil {
+			fmt.Printf("Invalid host - please try again")
+			continue
+		}
+		hosts = append(hosts, host)
+
+		more := false
+		prompt := &survey.Confirm{
+			Message: "Add another?",
+		}
+		survey.AskOne(prompt, &more, nil)
+		if !more {
+			break
+		}
+	}
+
+	return &baremetal.Platform{
+		LibvirtURI:         libvirtURI,
+		IronicURI:          ironicURI,
+		APIVIP:             apiVIP,
+		IngressVIP:         ingressVIP,
+		ExternalBridge:     externalBridge,
+		ProvisioningBridge: provisioningBridge,
+		Hosts:              hosts,
+	}, nil
+}
+
+// uriValidator validates if the answer provided in prompt is a valid
+// url and has non-empty scheme.
+func uriValidator(ans interface{}) error {
+	return validate.URI(ans.(string))
+}
+
+func apiVIPValidator(ans interface{}) error {
+	if ans.(string) != baremetaldefaults.APIVIP {
+		return validate.IP(ans.(string))
+	}
+	return nil
+}
+
+func ingressVIPValidator(ans interface{}) error {
+	if ans.(string) != baremetaldefaults.IngressVIP {
+		return validate.IP(ans.(string))
+	}
+	return nil
+}
+
+// interfaceValidator validates if the answer provided is a valid network
+// interface.  net.Interfaces in Go does not let us know if it's a bridge, but
```

Comment:
extra space.

Reply:
Being fixed in https://github.com/openshift-metal3/kni-installer/pull/143